src/core/system_parameters/gnss_satellite.cc

```diff
@@ -405,6 +404,24 @@ std::string Gnss_Satellite::what_block(const std::string& system_, uint32_t PRN_
                 case 32:
                     block_ = std::string("IIF");  // Plane F
                     break;
+                case 120:
```

Comment:
What you did here is consistent with the legacy code (good) but I don't really love it. This validation is begging for a database of some sort, even if it's just a map of satellites per constellation that is either initialized from a file or in an initializer list. This 400 lines of code is unwarranted complexity. 

The question of when to refactor is a subjective one, and it best comes with experience. You did well to follow the pattern, but I hope you had to hold your nose while you did it.